sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -148,6 +149,20 @@ class usm_allocator {
     }
   }
 
+  template <class U, usm::alloc AllocKindU, size_t AlignmentU>
+  friend bool operator==(const usm_allocator<T, AllocKind, Alignment> &One,
```

Comment:
These should probably be added to [sycl/doc/extensions/USM/USM.adoc](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/USM/USM.adoc)